### ISSUE TITLE
No Public access to Deserializer for Synapse Resources

### ISSUE THREAD

**Issue by manuchandel** [Azure/azure-sdk-for-net#18779]
Consider the use case:

1. We have all the sparkJobDefinition resources in Json files.
2. We have written a tool which publishes all of these resources to synapse workspace using  .net solution. (because there is no arm endpoint to publish pipelines, jobdefinitions, triggers (datafactory used to have these endpoints)).

We do not have access to these deserializers which can create objects from our json files. 

I don't see why these should not be publicly accessible methods to reference in nuget.

Please make these deserializers public.

Currenlty we need them for publishing jobdefinitions, pipelines, triggers 

https://github.com/Azure/azure-sdk-for-net/blob/b930426f410c66af97bddc875f2b2734c8a45469/sdk/synapse/Azure.Analytics.Synapse.Artifacts/src/Generated/Models/SparkJobDefinitionResource.Serialization.cs#L23

**Comment by jsquire** [comment]
Thank you for your feedback.  Tagging and routing to the team member best able to assist.

**Comment by idear1203** [comment]
@pakrym @chamons , here we got another user feedback to expose deserializers. Could you please help me understand why we don't expose deserializers in SDK? Thanks in advance.

**Comment by pakrym** [comment]
We haven't implemented serialization support for all models because most SDKs don't need it (clients serialize requests as part of client method calls) and it opens up another area where we need to keep backward compatibility.

Having said that we recognize that some SDKs need to be able to serialize/deserialize models outside of clients. That's why we recently added a feature that enables it. @ShivangiReja would be able to help you with enabling it for your models.

**Comment by ShivangiReja** [comment]
Hi @manuchandel, in order to access Serialization and Deserialization methods we have added json converter to models. It will forward calls to our existing Serialization and Deserialization methods.

It is not by default enabled. We have 3 options to enable this feature:

* Add `"x-csharp-usage": "converter"` property in your swagger. Check an example [here](https://github.com/Azure/autorest.csharp/blob/feature/v3/test/TestProjects/ModelWithConverterUsage/ModelWithConverterUsage.json#L93).
* Add `[CodeGenModel(Usage = new[] { "converter"})]` customization to your model class. Check an example [here](https://github.com/Azure/autorest.csharp/blob/feature/v3/test/TestProjects/ModelWithConverterUsage/Customized/ModelStruct.cs#L8).
* Add transform to your autorest.md. Something like this:
``` yaml
directive:
  from: swagger-document
  where: $.definitions.ModelClass
  transform: >
    $["x-csharp-usage"] = "converter";
````


After enabling this, your generated code would look something like [this](https://github.com/Azure/autorest.csharp/blob/feature/v3/test/TestProjects/ModelWithConverterUsage/Generated/Models/ModelClass.Serialization.cs#L15) and [here](https://github.com/Azure/autorest.csharp/blob/feature/v3/test/AutoRest.TestServer.Tests/ModelWithConverterUsageTests.cs#L14-L35) is the usage.

**Comment by ShivangiReja** [comment]
Hi @manuchandel, in order to access Serialization and Deserialization methods, we have added json converter to models. It will forward calls to existing Serialization and Deserialization methods.

This feature is not enabled by default. We have 3 options to enable this feature:
* Add `"x-csharp-usage": "converter"` property in your swagger. Check an example [here](https://github.com/Azure/autorest.csharp/blob/feature/v3/test/TestProjects/ModelWithConverterUsage/ModelWithConverterUsage.json#L93).
* Add `[CodeGenModel(Usage = new[] { "converter"})]` customization for your model class. Check an example [here](https://github.com/Azure/autorest.csharp/blob/feature/v3/test/TestProjects/ModelWithConverterUsage/Customized/ModelStruct.cs#L8).
* Add transform to your autorest.md. Something like this:
``` yaml
directive:
  from: swagger-document
  where: $.definitions.ModelClass
  transform: >
    $["x-csharp-usage"] = "converter";
````

After you enable this feature, your generated code would look something like [this](https://github.com/Azure/autorest.csharp/blob/feature/v3/test/TestProjects/ModelWithConverterUsage/Generated/Models/ModelClass.Serialization.cs#L15) and [here](https://github.com/Azure/autorest.csharp/blob/feature/v3/test/AutoRest.TestServer.Tests/ModelWithConverterUsageTests.cs#L13-L35) is how you can use it.

**Comment by manuchandel** [comment]
I guess the synapse team can use the information provided by @ShivangiReja and decide if they want to add the converters.

**Comment by ShivangiReja** [comment]
@chamons could you please help @manuchandel and add converter to the models?

**Comment by idear1203** [comment]
Awesome. I will do the change based on the above information.

**Comment by chamons** [comment]
@idear1203 - I believe you resolved this in #19423. Can this be closed?

**Comment by idear1203** [comment]
@chamons - thanks for the reminder. @manuchandel  has confirmed this works from his side. So we can close this issue
